### PR TITLE
Tul 199/citace pro iframe in item view

### PR DESCRIPTION
**Pull request by HuynhKhoa1601**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |
## Problem description
https://github.com/dataquest-dev/dspace-angular/issues/199

### CODE REVIEW

File: src/app/core/shared/tul/constants.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+//Third party API
+export const CITACE_PRO_BASE_URL = 'https://www.citacepro.com/api/dspace/citace/oai';
```
Comment (milanmajchrak) [inline src/app/core/shared/tul/constants.ts:2]: This is only reminder: this constants will be retrieved from BE (clarin-dspace.cfg)

File: src/app/item-page/simple/field-components/specific-field/citation/item-page-citation.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { ItemPageCitationFieldComponent } from './item-page-citation.component';
+import { DomSanitizer } from '@angular/platform-browser';
+import { ChangeDetectionStrategy, NO_ERRORS_SCHEMA } from '@angular/core';
+
+describe('ItemPageCitationFieldComponent', () => {
+  let component: ItemPageCitationFieldComponent;
+  let fixture: ComponentFixture<ItemPageCitationFieldComponent>;
+  let sanitizer: DomSanitizer;
+  const mockHandle = '15240/151431';
```
Comment (milanmajchrak) [inline src/app/item-page/simple/field-components/specific-field/citation/item-page-citation.component.spec.ts]: Is it a real handle? I think for test purposes we should better use some fake data :)

File: src/app/item-page/simple/field-components/specific-field/citation/item-page-citation.component.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Component, Input } from '@angular/core';
+import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
+import {CITACE_PRO_BASE_URL, UNIVERSITY_USING_DSPACE} from '../../../../../core/shared/tul/constants';
```
Comment (milanmajchrak) [inline src/app/item-page/simple/field-components/specific-field/citation/item-page-citation.component.ts]: empty space is missing in the brackets

**Review by milanmajchrak — CHANGES_REQUESTED** [review]

**Comment by milanmajchrak** [comment]
@HuynhKhoa1601 Please move this iframe CitacePRO component into this component body: 
image

**Comment by HuynhKhoa1601** [comment]
thanks I'll update it

**Comment by milanmajchrak** [comment]
@HuynhKhoa1601 Please add this <ds-item-page-citation-field [handle]="object?.handle"></ds-item-page-citation-field> component into `src/app/item-page/simple/item-types/publication/publication.component.html` also. Because it won't be seen in the publication.

**Review by milanmajchrak — APPROVED** [review]

**Pull request merged**